### DISCUSSION TITLE
Weights for objectives with NSGA-II algorithm?

### DISCUSSION THREAD

**Discussion by dinohsu1019** [anyoptimization/pymoo#519] [Q&A] [answered]
The following are mutually related questions around NSGA-II algorithm (possibly also other multi-objective optimization ones):
1. If one objective is more important than others, can I, say, take square of it as the objective, i.e., transformation of the objective?
2. Is there a set of weights that can be specified for each objective that drive the optimization? This is about the optimization process and the selection of non-dominated solutions, which is different from the set of weights specified to get the optimal solution from non-dominated solutions.
3. Does this set of weights even make sense? If not, what are the trade-offs among the objectives along the selection of non-dominated solutions? 
4. If one more independent objective converges faster and influence the exporation of the other three more mutually correlated objectives, do I scale the former one down say by a factor 10 to slow it down and allow others have time to explore more?
Please shed some light, thanks in advance.

**Comment by blankjul** [comment, marked as answer]
1. No. Each objective has the same importance in NSGA-II.
2. I recommend reading about MOO algorithms which use weights, e.g. https://ieeexplore.ieee.org/document/4358754
3. Yes see above
4. Uniform scaling helps in general. However, algorithms try to be scaling invariant.
